src/Exception/CustomException.php

```diff
@@ -0,0 +1,11 @@
+<?php
+
+class CustomException extends Exception {
+
+    public function errorMessage($error) {
+
+        include("views/error.php");
```

Comment:
Iš pagrindų blogai. Exception objektai skirti saugoti informaciją apie klaidą/išskirtinę situaciją. (saugoti info o ne vykdyti kažkokį funkcionalumą, labai panašiai kaip User entity). Paprastai sagoma kokia klaida, kur ir galbūt kodėl įvyko. Kad nemeluoju gali paskaityti čia: https://docs.oracle.com/javase/tutorial/essential/exceptions/definition.html (pripažinsiu pavyzdžiai/dokumentacija su PHP exceptions yra ne tokie informatyvūs). Tuo tarpu klaidos atvaizdavimas/programos stabdymas - visai ne exception objekto atsakomybė (tai turėtų būti vykdoma ten kur "sugaunamas" exception objektas). Bandyk pats dar pasiskaityti ir perdaryti exceptions naudojimą.

Reply:
Taip... tu iš tiesų teisus apie exception objektą.. Dabar kilo klausimas ar tos CustomException klasės tikrai reikia, nes realiai informacijos apie klaidą ekrane tikrai nevaizduosiu (net ir skaičiau , kad vartotojui to matyti negalima) , o rodysiu tik savo atvaizdavimo failą. Tu savo užduotyje nurodei  " Manau kad rezultate turėtų būti vienas failas iš kurio kviečiamas error template." tai galbūt tos klasės čia reikia, bet gal tu norėjai, kad padaryčiau kokį klaidų žurnalą (pvz kokiam txt faile ar duombazėje) , kur būtų registruojamos visos klaidos (kažką jau buvai pasakojęs dėl šito kai buvom susitikę) arba gal dar kažką praleidau ir ta error informacija objekte turėtų būti dar kitaip panaudojama. Tai va, jei nesunku, užvesk ant kelio, ką norėtum pamatyti galutiniam rezultate, nes panašu jog susimėčiau čia...

P.S dėkui dėl merginimo papildymo, akivaizdu ir aš pritingiu, kai kalba pasisuka apie pavadinimus :D

Reply:
Taip, dažnai vartotojui detalaus klaidos pranešimo matyti nereikia/negalima. Bet pvz vietoje "Užsiregistruoti nepavyko, toks vartotojo vardas jau egzistuoja" parodžius standartinį "Įvyko klaida" vartotojas liks nesupratęs kame šuo pakastas. Tai kadangi pats apie tai prakalbai - padarom taip: apsirašyk 2 exceptionų tipus/klases - vieną tokioms klaidoms kurias užteks tik parodyti vartotojui nepakeistas, o kitą "jautrioms" klaidoms, kur vartotojas matytų tik "Įvyko klaida" (arba ką nors kito gana abstraktaus). Turėk omeny, kad klasės pavadinimas `CustomException` nelabai informatyvus. Gaudyk apsirašytus exception viename faile (pasigalvok kuriame gudriausia) ir vienu atveju klaidos puslapyje rodyk abstraktų pranešimą (šiuo atveju tikroji klaida nueis į užmarštį, niekas jos nepamatys - tai pataisysime jau netrukus), o kitu - konkretų klaidos pranešimą.
Jei dar turi klausimų - prašom klaust. Pasiekėm tokį lygį kai ir išgooglinti ne taip jau paprasta :smile:

Reply:
Taip truputį aiškiau ko iš manęs nori, vis dėl to, vis dar yra jausmas, kad kažko nepagaunu ir dar bjauriau, kad jaučiu, jog tai net nėra kažkas sudėtingo. Padariau commit'ą padarydamas maždaug kaip manau jog turėtų būt. Tiesa, neįsivaizduoju, ką būtų galima rašyti į PublicExceptions klasę , nes realiai ta pats $errorMessage kintamasis ir tinka kaip error pranešimas (dėl to ten ties ta vieta tiesiog pakartojau kintamąjį be reikalo, žinau, kad ten nesamonė), juk nevaizduosim ekrane getMessage(), getFile() getLine() funkcijų. Galbūt toje klasėje saugoti, visas reikiamas $error žinutes (kaip pvz "slaptažodis per trumpas" , "slaptažodis neteisingas" ir panašiai , tuomet ta klasės būtų kaip getteris,  bet man pačiam toks variantas neatrodo labai logiškas. Kiek žiūrėjau į pavyzdžius tai dauguma visi panašūs pvz: https://www.w3schools.com/php/php_exception.asp arba https://www.geeksforgeeks.org/exception-handling-in-php/ , bet gal šitie pavyzdžiai nevisai man tinka ir per daug mane blaško ? Įdomu ką pasakysi, peržvelgęs mano commitą.